typescript/29-allergies/allergies.ts

```diff
@@ -0,0 +1,43 @@
+type Allergen =
+  | 'eggs'
+  | 'peanuts'
+  | 'shellfish'
+  | 'strawberries'
+  | 'tomatoes'
+  | 'chocolate'
+  | 'pollen'
+  | 'cats';
+
+export class Allergies {
+  private readonly score: number;
+  private readonly allergens: Allergen[];
+
+  constructor(allergenIndex: number) {
+    this.score = allergenIndex;
+
+    this.allergens = [
+      'eggs',
+      'peanuts',
+      'shellfish',
+      'strawberries',
+      'tomatoes',
+      'chocolate',
+      'pollen',
+      'cats',
+    ]
+  }
+
+  public list(): Allergen[] {
+    const result: Allergen[] = [];
+    for (let i = 0; i < this.allergens.length; i++) {
+      if ((this.score & (1 << i)) !== 0) {
+        result.push(this.allergens[i]);
+      }
+    }
+    return result;
+  }
+
+  public allergicTo(allergen: Allergen): boolean {
```

Comment:
Add a new function
```ts
export function calculateScore(allergens: Allergen[]): number
```
that would work out score from known list of affecting allergens.
Add comprehensive test coverage for it.